lib/std/zig/render.zig

```diff
@@ -2377,6 +2379,28 @@ fn renderSpace(ais: *Ais, tree: Ast, token_index: Ast.TokenIndex, lexeme_len: us
     }
 }
 
+// If the symbol is escaped with @"" syntax, only render it that way if it
+// contains invalid characters, otherwise render it with canonical syntax.
+fn renderCanonicalSymbol(gpa: Allocator, ais: *Ais, tree: Ast, token_index: Ast.TokenIndex, space: Space) Error!void {
+    const lexeme = tokenSliceForRender(tree, token_index);
+    if (lexeme[0] == '@') {
+        const symbol = std.zig.string_literal.parseAlloc(gpa, lexeme[1..]) catch |err| switch (err) {
+            error.InvalidLiteral => unreachable,
+            else => |e| return e,
+        };
+        defer gpa.free(symbol);
+        if (std.zig.isValidId(symbol)) {
+            // Canonicalize symbol that was unnecessarily escaped with @""
+            try ais.writer().writeAll(symbol);
+        } else {
+            try ais.writer().writeAll(lexeme);
+        }
+    } else {
+        try ais.writer().writeAll(lexeme);
+    }
+    try renderSpace(ais, tree, token_index, lexeme.len, space);
+}
+
```

Comment:
Hmm, I wonder if we could just check in `renderToken()` if the token has tag `identifier` and apply this logic there if so. This would avoid needing to update all these call sites to call the new function and potentially forgetting one. It would also be more robust to future code churn.